File: Backend/Movie/src/main/java/com/cts/movie/model/ShowTime.java
package com.cts.movie.model;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class ShowTime {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "movieId")
	private Movie movie;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "theatreId")
	private Theatre theatre;

	private String showTime;

	private float pricePerTicket;

	private int totalNoOfTickets;

	private int availableTickets;

	private List<String> noOfTicketsBooked;

	private String ticketStatus;

}

File: Frontend/MovieBookingAppUi/src/app/showTime/show-time.ts
import { Movie } from "../_classes/movie";
import { Theatre } from "../_classes/theatre";


export class ShowTime {
    constructor(public id: number, public movie: Movie, public theatre: Theatre, public showTime: any, public pricePerTicket: number,
        public totalNoOfTickets: number, public availableTickets: number,
        public noOfTicketsBooked: string[], public ticketStatus: string) { }
}
